### ISSUE TITLE
Tiered Retention Times

### ISSUE THREAD

**Issue by MrAnonymous722** [blakeblackshear/frigate#2590]
**Describe what you are trying to accomplish and why in non technical terms**
I would like to be able to set multiple retention rules per camera to sort of "trim down" a recording as it gets older. This would allow me to keep all recordings from a camera even with no motion for x days, then trim it down to only motion segments and to keep those for another period of time before finally only keeping segments that are needed to meet the clip's retention policy. 

**Describe the solution you'd like**
The ability to set multiple retain modes on a single camera. Or perhaps to require them all but need to set the days to retain each type (all, motion, active object)

**Describe alternatives you've considered**
Only using retain all, but for a short period of time due to disk space constraints.

**Comment by blakeblackshear** [comment]
Have you seen the changes in 0.10.0? It's getting pretty close to this.

**Comment by MrAnonymous722** [comment]
Yes, that’s pretty much what this is inspired by. I love that I can have
video retained by motion and active objects, however I was hoping I could
for example have all segments retained for a day, then after that just the
motion then active objects then finally delete, if it’s no longer being
retained by an event.

**Comment by joeShuff** [comment]
I made a similar request so I've closed that one an I'll paste some info here that I feel may be suitable. 👍 

**Describe what you are trying to accomplish and why in non technical terms**
I want to be able to define multiple retain rules for a camera so that I can save storage space in the long run but not lose anything important. For example:
For 7 days keep all footage `record -> retain -> mode = all`.
For 14 days keep all motion footage `record -> retain -> mode = motion`.
For 31 days keep all events footage `record -> retain -> move = active_objects`.

My understanding is that everything in `active_objects` is a subset of everything in `motion` which is a subset of `all` so as time passes it will slowly get rid of "useless" footage to save space. 

**Describe the solution you'd like**
To be able to define multiple rules for retention for a cameras config. e.g.

```
camera:
  record:
      retain:
          - days: 7
            mode: all
          - days: 14
            mode: motion
          - days: 31
            mode: active_objects
```

**Describe alternatives you've considered**
I've considered just using `active_objects` for event retention and `all` for record but again this could lose some important data. Unfortunately I've had to wade through my footage of a camera to get evidence for the police and I was lucky that I set the `record` retention so high because it wasn't an event but would have been motion. However leaving `all` on record builds up a lot of useless (motionless) footage.

**Comment by felipecrs** [comment]
Or maybe:

```
camera:
  record:
      retain:
        continuous:
          days: 7
        motion:
          days: 14
        active_objects:
          days: 31
```

I find the word _continuous_ more intuitive than _all_.

**Comment by okdas** [comment]
I agree, `continuous` is the word by which I located this issue.

**Comment by cameronjust** [comment]
Really looking forward to this feature. I had asked on a few forums if this was possible and a number of people think it already works with this config.

```
record:
  enabled: true
  retain:
    days: 7
    mode: all
  events:
    pre_capture: 5
    post_capture: 10
    objects:
    - car
    - person
    - cat
    - bird
    retain:
      default: 30
      mode: motion
      objects:
        person: 40
        cat: 30
        car: 14
```

It definitely only keeps events beyond 7 days and all motion (with no event) is not there anymore.

**Comment by z87094** [comment]
I tried using the example in my config (running 0.16.2) and I'm getting the following error:

```
2025-11-25 08:30:42.727022567  *************************************************************
2025-11-25 08:30:42.727027515  *************************************************************
2025-11-25 08:30:42.727030347  ***    Your config file is not valid!                     ***
2025-11-25 08:30:42.727034267  ***    Please check the docs at                           ***
2025-11-25 08:30:42.727035757  ***    https://docs.frigate.video/configuration/          ***
2025-11-25 08:30:42.727037675  *************************************************************
2025-11-25 08:30:42.727039106  *************************************************************
2025-11-25 08:30:42.727064632  ***    Config Validation Errors                           ***
2025-11-25 08:30:42.727066400  *************************************************************
2025-11-25 08:30:42.727067279
2025-11-25 08:30:42.813088030  Line #  : 684
2025-11-25 08:30:42.813092636  Key     : record -> continuous
2025-11-25 08:30:42.813093926  Value   : {'days': 3}
2025-11-25 08:30:42.813095258  Message : Extra inputs are not permitted
2025-11-25 08:30:42.813096198
2025-11-25 08:30:42.813097185  Line #  : 686
2025-11-25 08:30:42.813098265  Key     : record -> motion
2025-11-25 08:30:42.813129866  Value   : {'days': 28}
2025-11-25 08:30:42.813131236  Message : Extra inputs are not permitted
2025-11-25 08:30:42.813132163
2025-11-25 08:30:42.813133482  *************************************************************
2025-11-25 08:30:42.813134905  ***    End Config Validation Errors                       ***
2025-11-25 08:30:42.813136292  *************************************************************
```

Here's what I have for my recording config:
```
record:
  enabled: True
  continuous:
    days: 3
  motion:
    days: 28
  alerts:
    retain:
      days: 30
      mode: active_objects
  detections:
    retain:
      days: 30
      mode: active_objects
```

Is there something I'm missing here? 

Thank you!

**Comment by hawkeye217** [comment]
Your config is invalid. See the official documentation: https://docs.frigate.video/configuration/record/

**Comment by NickM-27** [comment]
You're trying to use the 0.17 config format
